src/v/cloud_storage/segment_chunk_api.cc

```diff
@@ -425,4 +425,43 @@ std::unique_ptr<chunk_eviction_strategy> make_eviction_strategy(
     }
 }
 
+segment_chunk_range::segment_chunk_range(
+  const segment_chunks::chunk_map_t& chunks,
+  size_t prefetch,
+  chunk_start_offset_t start) {
+    auto it = chunks.find(start);
```

Comment:
nit: maybe assert that `it != chunks.end()` here

Reply:
added